pkg/execution/state/redis_state/queue.go

```diff
@@ -519,11 +535,14 @@ func (q QueuePartition) MarshalBinary() ([]byte, error) {
 // The queue score must be added in milliseconds to process sub-second items in order.
 func (q *queue) EnqueueItem(ctx context.Context, i QueueItem, at time.Time) (QueueItem, error) {
 	if len(i.ID) == 0 {
-		i.ID = ulid.MustNew(ulid.Now(), rnd).String()
+		i.SetID(ctx, ulid.MustNew(ulid.Now(), rnd).String())
+	} else {
+		// Hash the ID.
+		// TODO: What if this is already hashed?
```

Comment:
how do we know if there's a collision?

Reply:
This is a followup for later.  We're going to ignore it for now;  it's a QoL issue as we expand on the queue.